### PR TITLE
Fix for data race in MKLDNNSharedMemory::valid

### PR DESCRIPTION
**Pull request by vladislav-volkov**
### Details:
 - Fix for data race in MKLDNNSharedMemory::valid

### Tickets:
 - CVS-55643

### CODE REVIEW

File: inference-engine/src/mkldnn_plugin/mkldnn_weights_cache.cpp [hunk 1]
```diff
@@ -25,11 +25,11 @@ MKLDNNWeightsSharing::MKLDNNSharedMemory::operator MKLDNNMemoryPtr() const {
 }
 
 bool MKLDNNWeightsSharing::MKLDNNSharedMemory::isValid() const {
-    return memory->valid;
+    return memory->valid.load(std::memory_order_acquire);
 }
 
 void MKLDNNWeightsSharing::MKLDNNSharedMemory::valid(bool b) {
-    memory->valid = b;
+    memory->valid.store(b, std::memory_order_release);
```
Comment (maxnick) [inline inference-engine/src/mkldnn_plugin/mkldnn_weights_cache.cpp:32]: Do we really need any memory access synchronization other than the `valid` flag itself? If not, it seems that relaxed ordering should be enough.

Comment (maxnick) [reply inference-engine/src/mkldnn_plugin/mkldnn_weights_cache.cpp:32]: Discussed offline. We need such ordering indeed.

**Review by maxnick — APPROVED** [review]

File: inference-engine/src/mkldnn_plugin/mkldnn_weights_cache.cpp (hunk 1 above)
Comment (vladislav-volkov) [reply inference-engine/src/mkldnn_plugin/mkldnn_weights_cache.cpp:32]: All graphNode->execute(...) operations sholud be completed before MKLDNNSharedMemory::valid. So I think it is better to use memory_order_release.

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
@vladislav-volkov please resolve merge conflicts

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**